packages/playwright-core/src/utils/dependencies.ts

```diff
@@ -179,6 +179,28 @@ export async function validateDependenciesLinux(sdkLanguage: string, linuxLddDir
   }
 
   const maybeSudo = (process.getuid() !== 0) && os.platform() !== 'win32' ? 'sudo ' : '';
+  // Check if we are running in a docker container which version does not match.
+  // Ignore patch versions: we never roll browsers in patch releases.
+  if (process.env.__PLAYWRIGHT_DOCKER_DISPLAY_NAME && process.env.__PLAYWRIGHT_DOCKER_VERSION && !process.env.__PLAYWRIGHT_DOCKER_VERSION.startsWith(utils.getPlaywrightVersion(true /* majorMinorOnly */))) {
+    const pwVersion = utils.getPlaywrightVersion();
+    const requiredDockerImage = process.env.__PLAYWRIGHT_DOCKER_DISPLAY_NAME.replace(process.env.__PLAYWRIGHT_DOCKER_VERSION, pwVersion);
+    throw new Error('\n' + utils.wrapInASCIIBox([
+      `Host system is missing a few dependencies to run browsers.`,
+      ``,
+      `This is due to docker image version not matching Playwright version:`,
+      `- Playwright: ${pwVersion}`,
+      `-     Docker: ${process.env.__PLAYWRIGHT_DOCKER_VERSION}`,
+      ``,
+      `Either:`,
+      `- (recommended) use docker image "${requiredDockerImage}"`,
+      `- (alternative) run the following command inside docker to install missing dependencies:`,
+      ``,
+      `    ${maybeSudo}${buildPlaywrightCLICommand(sdkLanguage, 'install-deps')}`,
```

Comment:
I see "Unhappy path" below, so this command is not always the right one? Let's be careful.